common/src/types/v0/transport/replica.rs

```diff
@@ -21,6 +22,103 @@ impl GetReplicas {
     }
 }
 
+/// Pool on which the resides
+#[derive(Serialize, Default, Debug, Clone, Eq, PartialEq)]
+pub struct ReplicaPool {
+    /// name of the pool
+    pub name: PoolId,
+    /// uuid of the pool
+    pub uuid: Option<PoolUuid>,
+}
+
+impl ReplicaPool {
+    /// Return a new `Self` from the provided arguments
+    pub fn new(name: &PoolId, uuid: Option<PoolUuid>) -> Self {
+        Self {
+            name: name.clone(),
+            uuid,
+        }
+    }
+    /// get the uuid of the pool
+    pub fn get_uuid(&self) -> Result<PoolUuid, uuid::Error> {
+        match self.uuid.clone() {
+            Some(uuid) => Ok(uuid),
+            None => {
+                // TODO add a getter function to fetch the uuid of the pool
+                Ok(PoolUuid::try_from(self.name.to_string())?)
+            }
+        }
+    }
+
+    /// get name of the pool from uuid
+    pub fn get_name_from_uuid(_uuid: &PoolUuid) -> PoolId {
+        unimplemented!()
+    }
+}
+
+impl<'de> Deserialize<'de> for ReplicaPool {
```

Comment:
I think we don't need to implement this manually by making ReplicaPool as such:
```rust
pub enum PoolRef {
    Named { name: PoolId },
    Uuid { name: PoolId, uuid: PoolUuid },
}
```
And then on the replicaSpec use #[serde(untagged)]

Reply:
Done.